src/test/java/com/endava/internship/mocking/repository/InMemUserRepositoryTest.java

```diff
@@ -0,0 +1,31 @@
+package com.endava.internship.mocking.repository;
+
+import com.endava.internship.mocking.model.Status;
+import com.endava.internship.mocking.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class InMemUserRepositoryTest {
+    InMemUserRepository userRepository;
+
+    @BeforeEach
+    void setUp() {
+        userRepository = new InMemUserRepository();
+    }
+
+    @Test
+    void shouldFindUserByIdOrReturnEmpty() {
+        User testUser = new User(2, "Maria", Status.ACTIVE);
+
+        assertThat(userRepository.findById(2)).hasValue(testUser);
```

Comment:
It works. InMemUserRepository does not allow to save new Users, it just creates and saves a bunch of them in repository constructor. testUser was created just to avoid appearing a magic user in assertion.